homeassistant/components/motionblinds_ble/cover.py

```diff
@@ -0,0 +1,283 @@
+        self._device.register_connection_callback(self.async_update_connection)
+        self.entity_description = COVER_TYPES[entry.data[CONF_BLIND_TYPE]]
+        self.config_entry: ConfigEntry = entry
+
+        self._attr_name: str = device.display_name
+        self._attr_unique_id: str = entry.data[CONF_ADDRESS]
+        self._attr_device_info: DeviceInfo = DeviceInfo(
+            connections={(CONNECTION_BLUETOOTH, entry.data[CONF_ADDRESS])},
+            manufacturer=MANUFACTURER,
+            model=entry.data[CONF_BLIND_TYPE],
+            name=self._attr_name,
+        )
+
+    async def async_update(self) -> None:
+        """Update state, called by HA if there is a poll interval and by the service homeassistant.update_entity."""
+        _LOGGER.debug("(%s) Updating entity", self.config_entry.data[CONF_MAC_CODE])
+        await self._device.connect()
+
+    async def async_stop_cover(self, **kwargs: Any) -> None:
+        """Stop moving the blind."""
+        _LOGGER.debug("(%s) Stopping", self.config_entry.data[CONF_MAC_CODE])
+        await self._device.stop()
+
+    @callback
+    def async_update_running(
+        self, running_type: MotionRunningType | None, write_state: bool = True
+    ) -> None:
+        """Update whether the blind is running (opening/closing) or not."""
+        self._attr_is_opening = (
+            False
+            if running_type
+            in [None, MotionRunningType.STILL, MotionRunningType.UNKNOWN]
+            else running_type is MotionRunningType.OPENING
+        )
+        self._attr_is_closing = (
+            False
+            if running_type
+            in [None, MotionRunningType.STILL, MotionRunningType.UNKNOWN]
+            else running_type is not MotionRunningType.OPENING
+        )
+        if running_type is not MotionRunningType.STILL:
+            self._attr_is_closed = None
+        if write_state:
+            self.async_write_ha_state()
+
+    @callback
+    def async_update_position(
+        self,
+        position: int | None,
+        tilt: int | None,
+    ) -> None:
+        """Update the position of the motor."""
+        self._attr_current_cover_position = (
+            100 - position if position is not None else None
+        )
+        self._attr_current_cover_tilt_position = (
+            100 - round(100 * tilt / 180) if tilt is not None else None
+        )
+        self._attr_is_closed = (
+            self._attr_current_cover_position == 0 if position is not None else None
+        )
+        self.async_write_ha_state()
+
+    @callback
+    def async_update_connection(self, connection_type: MotionConnectionType) -> None:
+        """Update the connection status."""
+        # Write state to update connection state attribute
+        self.async_write_ha_state()
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, str]:
+        """Return the state attributes."""
+        return {ATTR_CONNECTION: self._device.connection_type}
```

Comment:
device.connection_type is the `MotionConnectionType`, which is either `Connected`, `Connecting`, `Disconnected` or `Disconnecting`. This information is displayed as an attribute and in later also in a sensor. By using `BleakClient.is_connected` you cannot determine whether or not the blind is already being connected to. Also, you can register a callback for the `MotionConnectionType`

Reply:
Alright, this should be a seperate sensor entity. Therefore I suggest removing it from this initial PR and adding it as a seperate PR as a sensor entity once this PR is merged.

Reply:
I think the state attribute is still useful as it is an addition to the sensor entity and it makes it easier to see if the blind is connected (even if the sensor entity is not present) as it is available right underneath the cover in the user interface

Reply:
Having extra_state_attributes means those extra attributes and the main state both get saved in the database each time either the state_attribute or the main state changes. Therefore for optimal performance I think it is discuraged to have extra_state_attributes.

Reply:
Is there another way of mentioning the state underneath the cover in the user interface without it saving it to the database? I could remove it but then you currently have no way of knowing if the blind is connected, disconnected or if you are already connecting to it

Reply:
Well we will when the sensor PR is implemented and merged.

If you really want to keep it, I can ask for a second oppinion about it.
(I will probably want a second oppinion anyway because I have not merged a completly new integration before).

Reply:
Sure let's keep it for now and wait for the second opinion to decide, I'll leave this change open. Personally I think it's nice to see connection status underneath the cover as that makes it easier to see what is going on without having to constantly close the cover UI window to see the connection status using the sensor entity

Reply:
We are removing extra state attributes more and more so i think it's a logical step to move this to a sensor entity

Reply:
@joostlek Sure, I removed it. That does mean there is no way to see if the blind is connected without using the sensor entity though